libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -769,9 +769,9 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 		if (_cleanupNeeded)
 			m_context << Instruction::ISZERO << Instruction::ISZERO;
 		break;
-	case Type::Category::Function:
+	default:
 	{
-		if (targetTypeCategory == Type::Category::Integer)
+		if (stackTypeCategory == Type::Category::Function)
```

Comment:
I think this should be `stackTypeCategory == Type::Category::Function && targetTypeCategory == Type::Category::Integer` as it checks for function type to address conversion.